mycli/sqlexecute.py

```diff
@@ -26,8 +26,11 @@ class SQLExecute(object):
 
     version_query = '''SELECT @@VERSION'''
 
+    # System variables cannot be SELECTed (e.g. @@version_comment)
+    # this MySQL bug was fixed on `5.0.22`
+    # Bug #15684 https://bugs.mysql.com/bug.php?id=15684
     version_comment_query = '''SELECT @@VERSION_COMMENT'''
-    version_comment_query_mysql4 = '''SHOW VARIABLES LIKE "version_comment"'''
+    version_comment_query_alter = '''SHOW VARIABLES LIKE "version_comment"'''
```

Comment:
Does the alternative syntax work in all mysql versions, including versions `<5.0.22`? If so, why don't we always use the alternative versions, and remove the version check altogether? Seems to me like it would make things a lot simpler.

Reply:
Hi @j-bennet 

It's the best practice to query the version comment by selecting the system variable, that is the statement `SELECT @@VERSION_COMMENT`, but just the history issue the bug caused the problem that we need to use the `SELECT ... LIKE ...`, so nothing but also we can use the alternative statement for all situations.

Thanks and regards

Reply:
> best practice

but why, is there any reason it is preferred?

Reply:
Is it `SELECT @@VERSION_COMMENT` better than `SHOW VARABLES LIKE "VERSION_COMMENT"`? Not only for the simpler, but also no the `LIKE` comparison.

Reply:
Simpler yes, but we don't care that much about simplicity of predefined automated queries. And  `LIKE` is not a requirement, you can do `SHOW VARIABLES WHERE Variable_name = 'version_comment'`

Reply:
Yes, and the `SELECT @@VERSION_COMMENT` is also faster than `SHOW VARIABLES WHERE/LIKE ...`, but this totally depends on your choie, it just for your reference, the most important thing is to fix the fatal bug quickly.